README.md

```diff
@@ -198,3 +198,92 @@ func Function(
 	
 }
 ```
+
+### Sync Tool and Operator Types
+
+Previously, packages relied on importing the types from individual operators and adding them as dependencies. This has led to eco-goinfra having many dependencies, making it harder to reuse and increasing the chance of version conflicts between indirect dependencies.
+
+By using just the runtime client and removing dependencies on individual operators, the sync tool reduces the number of eco-goinfra dependencies and makes packages more modular. It will periodically copy over types from operator repos to this repo by cloning the desired path in the operator repo, applying changes specified in the config file, and then copying the resulting types into this repo.
+
+#### Running
+
+To run the sync tool, use the `lib-sync` makefile target.
+
+```
+make lib-sync
+```
+
+It may be faster to sync only one config file instead using the `--config-file` flag.
+
+```
+go run ./internal/sync --config-file ./internal/sync/configs/<config-file.yaml>
+```
+
+If the sync fails while adding a new set of operator types, remove the synced directory from `schemes/<pkg-to-sync>` and rerun the sync.
+
+#### Configuration
+
+Config files for the sync tool live in the [internal/sync/configs](./internal/sync/configs/) directory. A good example of all the features available is in the [nvidia-config.yaml](./internal/sync/configs/nvidia-config.yaml) file.
+
+Each config is a YAML document with a top level list of objects that follow this form:
+
+```yaml
+- name: operator-repo # 1
+  sync: true # 2
+  repo_link: "https://github.com/operator/repo" # 3
+  branch: main # 4
+  remote_api_directory: pkg/apis/v1 # 5
+  local_api_directory: schemes/operator/operatortypes # 6
+  replace_imports: # 7
+    - old: '"github.com/operator/repo/pkg/apis/v1/config"' # 8
+      new: '"github.com/openshift-kni/eco-goinfra/pkg/schemes/operator/operatortypes/config"'
+    - old: '"github.com/operator/repo/pkg/apis/v1/utils/exec"'
+      new: exec "github.com/openshift-kni/eco-goinfra/pkg/schemes/operator/operatortypes/executils" # 9
+  excludes:
+    - "*_test.go" # 10
+```
+
+1. Name, which does not have to be unique, identifies the repos in logs and controls where it is cloned during the sync. It does not affect the outcome of the sync itself and can be set as desired.
```

Comment:
Personally I think we should require that names only contain alphanumeric characters and hyphens. I noticed weird behavior with removing cloned dirs when using forward slashes. I think spaces are okay but we're better off keeping it simple.